lib/db/schema/patch-045-046.sql

```diff
@@ -0,0 +1,224 @@
+-- Add emails table and corresponding procedures to
+-- create, get and delete them.
+
+CREATE TABLE IF NOT EXISTS emails (
+  normalizedEmail VARCHAR(255) NOT NULL PRIMARY KEY,
+  email VARCHAR(255) NOT NULL,
+  uid BINARY(16) NOT NULL,
+  emailCode BINARY(16) NOT NULL,
+  isVerified BOOLEAN NOT NULL DEFAULT FALSE,
+  isPrimary BOOLEAN NOT NULL DEFAULT FALSE,
+  verifiedAt BIGINT UNSIGNED,
+  createdAt BIGINT UNSIGNED NOT NULL,
+  INDEX `uid` (`uid`)
+) ENGINE=InnoDB;
+
+DROP procedure IF EXISTS `createEmail_1`;
+
+CREATE PROCEDURE `createEmail_1` (
+    IN `normalizedEmail` VARCHAR(255),
+    IN `email` VARCHAR(255),
+    IN `uid` BINARY(16) ,
+    IN `emailCode` BINARY(16),
+    IN `isVerified` TINYINT(1),
+    IN `isPrimary` TINYINT(1),
+    IN `verifiedAt` BIGINT UNSIGNED,
+    IN `createdAt` BIGINT UNSIGNED
+)
+BEGIN
+    -- Currently, can not add an email that is specified in the
+    -- accounts table, regardless of verification state.
+    SET @emailExists = 0;
+    SELECT COUNT(*) INTO @emailExists FROM accounts a WHERE a.normalizedEmail = normalizedEmail;
+    IF @emailExists > 0 THEN
+        SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 1062, MESSAGE_TEXT = 'Email already exists in accounts table.';
+    END IF;
```

Comment:
This will need to be done inside a transaction, otherwise there's a race condition where the above check succeeds, but someone concurrently registers that email address before we run the `INSERT` below.